app/models/vm_or_template.rb

```diff
@@ -918,7 +918,14 @@ def log_all_proxies(all_proxy_list, message)
   end
 
   def log_proxies_vm_config
-    "[#{log_proxies_format_instance(self)}] on host [#{log_proxies_format_instance(host)}] #{ui_lookup(:table => "storages").downcase} [#{storage.name}-#{storage.store_type}]"
+    msg = "[#{log_proxies_format_instance(self)}] on host [#{log_proxies_format_instance(host)}] "\
+    "#{ui_lookup(:table => "storages").downcase} "
+    msg += if storage
+             "[#{storage.name}-#{storage.store_type}]"
+           else
+             "No storage"
+           end
+    msg
```

Comment:
Style:

- Avoid the line continuation `\`...just make it a single line.
- Prefer `<<` over `+=`
- There's no need to append and then return the variable, it can be done in one shot

```ruby
  def log_proxies_vm_config
    msg = "[#{log_proxies_format_instance(self)}] on host [#{log_proxies_format_instance(host)}] #{ui_lookup(:table => "storages").downcase} "
    msg << storage ? "[#{storage.name}-#{storage.store_type}]" : "No storage"
  end
```